relay-sampling/src/lib.rs

```diff
@@ -355,8 +356,10 @@ mod tests {
         event: &Event,
         dsc: &DynamicSamplingContext,
         now: DateTime<Utc>,
-    ) -> Option<SamplingMatch> {
-        SamplingMatch::match_against_rules(config.rules_v2.iter(), Some(event), Some(dsc), now)
```

Comment:
so here the tests was (and still is) bypassing the rule verification. I'm not sure if this is intended behaviour or an oversight. You could say its more unit-y but, it also is less relevant.